### PR TITLE
Remove size-limit-action and use the cli

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

There have been some odd results with the sizelimit action where code removals appear to have increased the bundle size. This happened in https://github.com/Shopify/polaris/pull/8255#issuecomment-1414478156 which is only visible when viewing the edit history in the comment.

Screenshot 2023-02-03 at 11 28 37 am

The sizelimit action is not actively maintained and is on [version 4 of the library](https://github.com/andresz1/size-limit-action/blob/dd31dce7dcc72a041fd3e49abf0502b13fc4ce05/package.json#L45). The value of the action is that it fails a Pull Request and adds a comment to the PR showing the size difference.

The only reason this PR exists is because of the visibility the comments previously provided by the action. So maybe we should keep the GitHub action... 😭

### WHAT is this pull request doing?

This PR keeps the functionality of failing a pull request if it goes over the size limit. However we lose the comment functionality. So it will be hard to spot PR's that unnecessarily increase the bundle size.

- [x] Bumps sizelimit to the latest version
- [x] Removes the size-limit-action
- [x] Runs `size-limit` from the CLI in our CI GitHub action
- [x] Adds `"webpack": false` for CSS bundle to stop wrong numbers, because webpack inserts style-loader runtime (≈2 kB) into the bundle.

### CODE REVIEW

**Comment by BPScott** [comment]
With this we loose the comment details the current size and any size difference from the previous build. I'd be tempted to say that the size change ("you added X kb in this PR") is the most important piece of information provided - it makes it clear when theres an unexpected jump. Personally I'd want to keep that either by keeping the action or reimplementing it somehow if you're convinced that it is buggy.

**Comment by alex-page** [comment]
I think you are right @BPScott. I'll leave it open to see other peoples thoughts.

It doesn't seem to be buggy.. It is working just quite out of date. I did have a red herring here though..

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -59,3 +59,6 @@ jobs:
 
       - name: Test
         run: yarn test
+
+      - name: Size limit
+        run: yarn size-limit
```
Comment (sam-b-rose) [inline .github/workflows/ci.yml:64]: I could see maybe adding a comment response if this fails. This way we get the CI failure and comment feedback. No required imo but something we could improve on if we wanted

**Review by sam-b-rose — APPROVED** [review]

**Review by laurkim — APPROVED** [review]
I wonder if that screenshot from the PR was just a blip. I double checked one of my PRs that reduced size limit and the size didn't go up and then down in an edited comment. But maybe that's because I never pushed up to my branch again after opening the PR.

I do think there's still value in having comments and agree with @samrose3's suggestion about having a comment if it fails as a nice to have

**Comment by alex-page** [comment]
Going to close this PR. When I opened it I saw some strange behaviour. I hope we don't need to reopen it and add back the comment logic.

**Pull request closed without merging**